File: workspaces/ws_001_tipcalcdeluxe/android/app/src/main/java/com/appfactory/tipcalcdeluxe/model/Rounding.kt
package com.appfactory.tipcalcdeluxe.model

import kotlin.math.ceil
import kotlin.math.floor

/** Rounding mode applied to the grand total of the bill. */
enum class Rounding(val label: String) {
    NONE("Exact"),
    DOWN("Round Down"),
    UP("Round Up");

    /** Apply this rounding mode to a grand total. */
    fun apply(total: Double): Double = when (this) {
        NONE -> total
        DOWN -> floor(total)
        UP -> ceil(total)
    }
}

File: workspaces/ws_001_tipcalcdeluxe/ios/TipCalculatorDeluxe/Models/Rounding.swift
import Foundation

/// Rounding mode applied to the final total bill.
enum Rounding: String, CaseIterable, Identifiable, Codable {
    case none
    case down
    case up

    var id: String { rawValue }

    /// Short label shown in the segmented control.
    var label: String {
        switch self {
        case .none: return "Exact"
        case .down: return "Round Down"
        case .up: return "Round Up"
        }
    }

    /// Apply the rounding to a grand total (in currency units).
    func apply(to total: Double) -> Double {
        switch self {
        case .none: return total
        case .down: return total.rounded(.down)
        case .up: return total.rounded(.up)
        }
    }
}
